ddtrace/tracer/tracer.go

```diff
@@ -367,4 +374,7 @@ func (t *tracer) sample(span *span) {
 		}
 		span.Metrics[sampleRateMetricKey] = rs.Rate()
 	}
+	if t.config.prioritySampling != nil {
```

Comment:
I may miss the flow (GitHub is not helping me with that) but I see:
```go
// apply applies sampling priority to the given span. Caller must ensure it is safe
// to modify the span.
func (ps *prioritySampler) apply(spn *span) {...}
```

and this caller seems to not lock the `span`? in the case of https://github.com/DataDog/dd-trace-go/pull/371/files#diff-a72827d70b740ff43d8e5a0dc6c4cb4fR366 it seems we lock the span (to update the `span.Metrics` map).

Do we need to do something similar, considering we're accessing the `span.Meta` in inner calls?

Reply:
Also fine here because sample is only called in startSpan, when the span is created.

Reply:
OK thanks for the explanation!